client/tests/app.spec.js

```diff
@@ -41,27 +45,37 @@ describe('App', () => {
       ]);
     });
 
-    it('changes the listingId state to be the number passed in as props', done => {
+    it('changes the listingId state to be the number passed in as props', async () => {
       const wrapper = mount(<App listingId={123456789}/>);
-      axios.get()
-        .then(() => {
-          setImmediate(() => {
-            wrapper.update();
-            const instance = wrapper.instance();
-            expect(instance.houseData.id).toEqual(123456789);
-            done();
-          });
+      const instance = wrapper.instance();
+
+      expect(instance.props.listingId).toEqual(123456789);
+      await Promise.all([
+        expect(axios.get).toHaveBeenCalledTimes(1),
+        wrapper.setState({houseData: {
+          id: 123456789,
+          listing_price: 123456,
+          name: 'Mamba'
+        }}, () => {
+          expect(wrapper.state().houseData.id).toEqual(123456789);
+          expect(wrapper.state().houseData.listing_price).toEqual(123456);
+          expect(wrapper.state().houseData.name).toEqual('Mamba');
         })
+      ])
     });
 
     // to see if rerendering works... will change as components are added
     it('renders the name of the agent to dom', async () => {
         const wrapper = mount(<App listingId={123456789}/>);
-
-        const instance = await wrapper.instance();
-        wrapper.setState({listingId: instance.houseData.id}, () => {
-          expect(wrapper.find('p').text()).toEqual('Mamba');
-        })
+        await Promise.all([
+          wrapper.setState({houseData: {
+            id: 123456789,
+            listing_price: 123456,
+            name: 'Mamba'
+          }}, () => {
+            expect(wrapper.find('p').text()).toEqual('Mamba');
+          })
+        ]);
```

Comment:
This test will be very brittle, it will likely break later in this since the p may be refactored into a <div> or other React component. One way to do this is by adding an Id to the selector since that can be kept across different components. Also you can use the .toHaveText() if you might be later adding other text around this component.

Reply:
Thanks for the suggestion! I guess this was more of a practice to see how to make rendering tests. I'll just take it out or skip it for now and get back to it when my app gets more complex